### PR TITLE
Get the IDR playbooks working smoothly with multiple deployments

### PR DESCRIPTION
**Pull request by manics**
Replaces `vm_prefix` with `idr_environment` in all openstack IDR playbooks, adds this as a group prefix in all IDR playbooks.

This means every server in an IDR deployment is in the following three groups
  1. one of `database-hosts` `omero-hosts` `gateway-hosts`
  2. `{{idr_environment}}-hosts`
  3. one of `{{idr_environment}}
-database-hosts` `{{idr_environment}}
-omero-hosts` `{{idr_environment}}
-gateway-hosts`

In practice this means it is possible to configure variables for
  1. all servers by purpose (database, omero, gateway)
  2. all servers in a given deployment (idr_environment)
  3. a server by purpose in a given deployment (intersection of the above)

At present (3) should only contain one server (which can be addressed as e.g. `{{idr_environment}}
-database-hosts[0]`, if we implement load-balancing there may be more than 1.

This also reverts some changes to `ansible.cfg` from https://github.com/openmicroscopy/infrastructure/pull/112 in order to get this repo working, these changes should be reverted in a `development` or `breaking` branch until everyone's happy.

`idr_environment` defaults to `idr` so that the IDR playbooks can be run without any private configuration. In most cases you should set `idr_environment` to something else and add matching group vars files. For example see `# IDR multinode omero` in Vagrantfile

Fixes the following from the config issue:

 * [x] tty required for sudo (causes `fatal: [demo3-omero]: FAILED! => {"failed": true, "msg": "Failed to get information on remote file (/etc/security/limits.d/95-omero.conf): MODULE FAILURE"}` (idr-omero.yml))
 * [x] use of `[0]` for which host to use is dangerous since it assumes one type per cloud.

Includes:
- https://github.com/openmicroscopy/infrastructure/pull/119 first commit only https://github.com/openmicroscopy/infrastructure/pull/119/commits/d25f4d15bf34504d362356a1fce9d94af5bf5591
- https://github.com/openmicroscopy/infrastructure/pull/120
- https://github.com/openmicroscopy/infrastructure/pull/123
- https://github.com/openmicroscopy/infrastructure/pull/125

For tidiness this PR should be rebased after merging the above three.

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#970](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/970/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/970/consoleText) for more details.
Possible conflicts:
  - PR #120 manics 'Fix Vagrantfile so `vagrant up` brings up all nodes'
    - ansible/Vagrantfile

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#971](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/971/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/971/consoleText) for more details.
Possible conflicts:
  - PR #120 manics 'Fix Vagrantfile so `vagrant up` brings up all nodes'
    - ansible/Vagrantfile

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#972](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/972/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/972/consoleText) for more details.
Possible conflicts:
  - PR #116 aleksandra-tarkowska 'prompt for jenkins password'
    - ansible/ansible.cfg
  - PR #120 manics 'Fix Vagrantfile so `vagrant up` brings up all nodes'
    - ansible/Vagrantfile
  - PR #121 manics 'Return non-zero error code if no hosts match'
    - ansible/ansible.cfg

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#973](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/973/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/973/consoleText) for more details.
Possible conflicts:
  - PR #120 manics 'Fix Vagrantfile so `vagrant up` brings up all nodes'
    - ansible/Vagrantfile

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#975](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/975/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/975/consoleText) for more details.
Possible conflicts:
  - PR #120 manics 'Fix Vagrantfile so `vagrant up` brings up all nodes'
    - ansible/Vagrantfile

**Comment by joshmoore** [comment]
--exclude

This is the combined state of the other PRs. Getting those in first.

**Comment by manics** [comment]
As discussed I've refactored `os-create.yml` into a role, and added an example playbook.

**Comment by manics** [comment]
I've disabled the default nginx header in https://github.com/openmicroscopy/infrastructure/pull/124/commits/c135ef474ae5a6fe1c93b74664e3525bba639691, this shouldn't be necessary since https://github.com/openmicroscopy/openmicroscopy/pull/4744 means the displayed host is generated on the client side, not the server.

**Comment by joshmoore** [comment]
Couple of questions:

 * time to remove "exclude"?
 * are the old top-level IDR playbooks slotted for removal?
 * if we need to ship our own os_volume for a while happy to start bringing these into infrastructure (as with https://github.com/ansible/ansible-modules-core/pull/4614)

**Comment by manics** [comment]
> time to remove "exclude"?

👍 

> are the old top-level IDR playbooks slotted for removal?

I'll remove them in a later PR once we know this works at EBI.

> if we need to ship our own os_volume for a while happy to start bringing these into infrastructure (as with ansible/ansible-modules-core#4614)

The main benefit was for testing the changes to infrastructure, in production it should be a one off copy so the time penalty doesn't matter so much.

File: ansible/README-os-idr.md [hunk 1]
```diff
@@ -10,84 +10,67 @@ Most of these scripts should also work on other platforms, providing the VM is b
...
+        -i inventory/openstack-private.py \
+        -u centos \
+        -e idr_environment=$IDR_ENVIRONMENT \
+        -e idr_nginx_ssl_self_signed=True \
+        -e ansible_ssh_common_args="'-o ProxyCommand=\\\"ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -W %h:%p -q centos@$BASTION_IP\\\" -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no'" \
+        idr-playbooks/os-idr-volumes.yml \
+        idr-playbooks/idr-dundee-nfs.yml \
+        idr-playbooks/idr-ebi-nfs.yml \
+        idr-playbooks/idr.yml \
+        idr-playbooks/idr-docker.yml
```
Comment (joshmoore) [inline ansible/README-os-idr.md]: This doesn't exist (yet?)

Comment (manics) [inline ansible/README-os-idr.md]: True...that's in my other branch

File: ansible/README-os-idr.md [hunk 2]
```diff
@@ -10,84 +10,67 @@ Most of these scripts should also work on other platforms, providing the VM is b
...
+Find the floating IP of the proxy/bastion server.
+Set `BASTION_IP` to the IP, and `IDR_ENVIRONMENT` to match the value from above.
+Run:
 
-Before running the playbook you must [setup your OpenStack environment variables](http://docs.openstack.org/user-guide/common/cli_set_environment_variables_using_openstack_rc.html).
-You can override variables at the command line, for example (note double quoting is necessary if spaces are present):
+    BASTION_IP=10.0.0.0
+    IDR_ENVIRONMENT=idr
+    ansible-playbook \
+        -i inventory/openstack-private.py \
```
Comment (manics) [inline ansible/README-os-idr.md]: This is the public repo so the instructions are designed to work as far as possible without any private config. I assumed any internal instructions would be elsewhere

Comment (joshmoore) [inline ansible/README-os-idr.md]: Understood, but in general, I think we may still need to re-evaluate openstack-private.py (later)

File: ansible/README-os-idr.md (hunk 1 above)
Comment (manics) [inline ansible/README-os-idr.md]: https://github.com/manics/infrastructure/tree/idr-terraform%2Bos-roles

**Comment by manics** [comment]
Disabled `omero_upgrade`

**Comment by joshmoore** [comment]
Shooting for having this to be in 0.3.0-rc1 (of IDR)

**Comment by joshmoore** [comment]
No objections in chat. Only non-IDR change is the selinux blocker. Merging.

**Pull request merged**